### PR TITLE
Add regen component

### PR DESCRIPTION
**Pull request by e-aakash**
Currently made changes so that existing system (Health, Damage) don't break.

Will add scheduled regen using this component.

### CODE REVIEW

File: src/main/java/org/terasology/logic/health/RegenComponent.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+package org.terasology.logic.health;
+
+import org.terasology.entitySystem.Component;
+import org.terasology.math.TeraMath;
+import org.terasology.network.Replicate;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class RegenComponent implements Component {
```
Comment (Cervator) [inline src/main/java/org/terasology/logic/health/RegenComponent.java:25]: I wonder if we should fully specify this as `RegeneratingComponent` - would be clearer (entity is actively regenerating) and leave more potential space if a later system want to mark entities with something like `RegenerableComponent`

File: src/main/java/org/terasology/logic/health/RegenComponent.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    private Map<String, Float> regenValue = new HashMap<>();
+    private Map<String, Long> regenEndTime = new HashMap<>();
+    public long lowestEndTime;
```
Comment (Cervator) [inline src/main/java/org/terasology/logic/health/RegenComponent.java:28]: I wonder if the wording could be improved from "lowest" here - to me at least it doesn't seem super intuitive vs the concepts of starting and ending things. `soonest` instead perhaps?

File: src/main/java/org/terasology/logic/health/RegenComponent.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    public boolean isEmpty() {
+        return regenValue.isEmpty();
+    }
+
+    public int getRegenValue() {
+        float totalValue = 0;
+        for (float value: regenValue.values()) {
+            totalValue += value;
+        }
+        if (totalValue < 0) {
```
Comment (Cervator) [inline src/main/java/org/terasology/logic/health/RegenComponent.java:77]: Maybe I'm mistaken but I have a feeling there should be _some_ sort of utility method either in Java Math or our Math lib that should be able to take a number and set it to 0 if it is indeed below 0 (or any arbitrary number). Pretty trivial though, but maybe if there's something obvious?

**Review by Cervator — APPROVED** [review]
Tested out locally and reviewed - looks pretty good overall I'd say, although it might be good with another set of eyes just in case. Found #9 but didn't expect multiplayer to be flawless yet. Going to merge as-is since I resolved some conflicts 👍

File: src/main/java/org/terasology/logic/health/RegenComponent.java (hunk 3 above)
Comment (syntaxi) [reply src/main/java/org/terasology/logic/health/RegenComponent.java:77]: `Math.max(totalValue, 0);`

Comment (syntaxi) [reply src/main/java/org/terasology/logic/health/RegenComponent.java:77]: `Math.max(totalValue, 0);`?

**Pull request merged**